Lib/typing.py

```diff
@@ -1552,7 +1592,12 @@ def __getitem__(self, params):
             params = (params,)
         msg = "Parameters to generic types must be types."
         params = tuple(_type_check(p, msg) for p in params)
-        _check_generic(self, params, self._nparams)
+        if not self._nparams:
+            raise TypeError(f"{self} is not a generic class")
+        actual_len = len(params)
+        if actual_len != self._nparams:
+            raise TypeError(f"Too {'many' if actual_len > self._nparams else 'few'} arguments for {self};"
+                            f" actual {actual_len}, expected {self._nparams}")
```

Comment:
small micro-optimisation:

```suggestion
        actual_len = len(params)
        if actual_len != self._nparams:
            if not self._nparams:
                raise TypeError(f"{self} is not a generic class")
            raise TypeError(f"Too {'many' if actual_len > self._nparams else 'few'} arguments for {self};"
                            f" actual {actual_len}, expected {self._nparams}")
```